### DISCUSSION TITLE
Irado

### DISCUSSION THREAD

**Discussion by csfolmer** [xirixiz/homeassistant-afvalwijzer#371] [Ideas]
Hi is there a possible way to add Irado?
https://www.irado.nl/afvalkalender

Found afvalinfo. but would like to use the nice cards of afvalkalender

**Comment by PatrickSt1991** [comment]
> Hi is there a possible way to add Irado?
> https://www.irado.nl/afvalkalender
> 
> Found afvalinfo. but would like to use the nice cards of afvalkalender

Do you have some info to login? Postal and housenumber?

Maybe there is a API available

**Reply by csfolmer** [reply]
3132SP 53A

Hope you can see a API ;-)

**Comment by PatrickSt1991** [comment]
@csfolmer i'm guessing luck is on your side today, their website is absolute shit :)

But luckily they have an android app which can be reversed engineered;
Response: 2.28s (Insomnia)
API: https://www.irado.nl/wp-json/wsa/v1/location/address/calendar/nextpickup?zipcode=3132SP&number=53&extention=A

```json
{
	"valid": true,
	"data": [
		{
			"date": "19\/02\/2025",
			"time": 1739923200,
			"day": 19,
			"month": 2,
			"year": 2025,
			"localAreaCode": "VL-GF7",
			"type": "gft"
		},
		{
			"date": "06\/03\/2025",
			"time": 1741219200,
			"day": 6,
			"month": 3,
			"year": 2025,
			"localAreaCode": "VL-PAP13",
			"type": "papier"
		}
	]
}
```

Guessing these are the first two pickups?

API: https://www.irado.nl/wp-json/wsa/v1/location/address/calendar/pickups?zipcode=3132SP&number=53&extention=A
Response: 2.21s (Insomnia)
JSON output [irando_pickups.json](https://github.com/user-attachments/files/18717574/irando_pickups.json) to big to show in a code box so thats why its a attachement.

This is only the API, so a pull request still needs to be created #390 to implement it in homeassistant-afvalwijzer. I will start working on it sometime today, but I can't provide an estimate for when it will be done.

**Reply by csfolmer** [reply]
> @csfolmer i'm guessing luck is on your side today, their website is absolute shit :)

haha
sorry for you ;-)

> But luckily they have an android app which can be reversed engineered; Response: 2.28s (Insomnia) API: https://www.irado.nl/wp-json/wsa/v1/location/address/calendar/nextpickup?zipcode=3132SP&number=53&extention=A

Yes they have a app, and great that you find a API!

> ```json
> {
> 	"valid": true,
> 	"data": [
> 		{
> 			"date": "19\/02\/2025",
> 			"time": 1739923200,
> 			"day": 19,
> 			"month": 2,
> 			"year": 2025,
> 			"localAreaCode": "VL-GF7",
> 			"type": "gft"
> 		},
> 		{
> 			"date": "06\/03\/2025",
> 			"time": 1741219200,
> 			"day": 6,
> 			"month": 3,
> 			"year": 2025,
> 			"localAreaCode": "VL-PAP13",
> 			"type": "papier"
> 		}
> 	]
> }
> ```
> 
> Guessing these are the first two pickups?

yes
![Screenshot 2025-02-08 10 21 21](https://github.com/user-attachments/assets/e03044a9-c09c-4ce7-82c6-0129280e9428)
screenshot of what I got now from afvalinfo

> API: https://www.irado.nl/wp-json/wsa/v1/location/address/calendar/pickups?zipcode=3132SP&number=53&extention=A Response: 2.21s (Insomnia) JSON output [irando_pickups.json](https://github.com/user-attachments/files/18717574/irando_pickups.json) to big to show in a code box so thats why its a attachement.
> 
> This is only the API, so a pull request still needs to be created #390 to implement it in homeassistant-afvalwijzer. I will start working on it sometime today, but I can't provide an estimate for when it will be done.

No problem! it's already great that you take the time to look into this and will take some time to implement it to afvalwijzer!
take you time, no rush.

**Comment by PatrickSt1991** [comment]
@csfolmer i've created a pull request for now i've only implemented the nextpickup api cause you already have that, in the comming days i'll work on embedding the pickups api aswell so that the entire year is possible instead of only two days.

It's now up to @xirixiz as the repo owner to check and merge if he finds it suitable

**Comment by PatrickSt1991** [comment]
@csfolmer If you don't wish to wait for the PR being merged you can use this fork, i've used to develop Irado and Cleaning Providers;
[Afvalwijzer with Irado and Cleaning Providers](https://github.com/PatrickSt1991/homeassistant-afvalwijzer/tree/main/custom_components/afvalwijzer)

Keep in mind this is still xirixiz his product, so you'll have to manually do everything, if you try to do it via HACS it won't download my fork :)

**Reply by csfolmer** [reply]
Thanks @PatrickSt1991 

Will try if i got the time!

But maybe @xirixiz will update his product soon .
Will save me some time to manually add yours😁

**Comment by PatrickSt1991** [comment]
@csfolmer de nieuwste versie die beschikbaar is bevat nu ook Irado

**Reply by csfolmer** [reply]
Ik zag het!
Hij is al geïnstalleerd alleen ff de plaatjes en automatisering ff goed zien te krijgen.

Bedankt voor de inzet
